cyw43/src/control.rs

```diff
@@ -78,64 +79,70 @@ impl<'a> Control<'a> {
-        self.set_iovar_u32("ampdu_mpdu", 4).await;
-        Timer::after(Duration::from_millis(100)).await;
-        //self.set_iovar_u32("ampdu_rx_factor", 0).await; // this crashes
+            self.set_iovar_u32("bus:txglom", 0).await;
+            Timer::after(Duration::from_millis(100)).await;
+            //self.set_iovar_u32("apsta", 1).await; // this crashes, also we already did it before...??
+            //Timer::after(Duration::from_millis(100)).await;
+            self.set_iovar_u32("ampdu_ba_wsize", 8).await;
+            Timer::after(Duration::from_millis(100)).await;
+            self.set_iovar_u32("ampdu_mpdu", 4).await;
+            Timer::after(Duration::from_millis(100)).await;
+            //self.set_iovar_u32("ampdu_rx_factor", 0).await; // this crashes
 
-        //Timer::after(Duration::from_millis(100)).await;
+            //Timer::after(Duration::from_millis(100)).await;
 
-        // evts
-        let mut evts = EventMask {
-            iface: 0,
-            events: [0xFF; 24],
-        };
+            // evts
+            let mut evts = EventMask {
+                iface: 0,
+                events: [0xFF; 24],
+            };
 
-        // Disable spammy uninteresting events.
-        evts.unset(Event::RADIO);
-        evts.unset(Event::IF);
-        evts.unset(Event::PROBREQ_MSG);
-        evts.unset(Event::PROBREQ_MSG_RX);
-        evts.unset(Event::PROBRESP_MSG);
-        evts.unset(Event::PROBRESP_MSG);
-        evts.unset(Event::ROAM);
+            // Disable spammy uninteresting events.
+            evts.unset(Event::RADIO);
+            evts.unset(Event::IF);
+            evts.unset(Event::PROBREQ_MSG);
+            evts.unset(Event::PROBREQ_MSG_RX);
+            evts.unset(Event::PROBRESP_MSG);
+            evts.unset(Event::PROBRESP_MSG);
+            evts.unset(Event::ROAM);
 
-        self.set_iovar("bsscfg:event_msgs", &evts.to_bytes()).await;
+            self.set_iovar("bsscfg:event_msgs", &evts.to_bytes()).await;
 
-        Timer::after(Duration::from_millis(100)).await;
+            Timer::after(Duration::from_millis(100)).await;
+          
+            // set wifi up
+            self.up().await;
 
-        // set wifi up
-        self.up().await;
+            Timer::after(Duration::from_millis(100)).await;
 
-        Timer::after(Duration::from_millis(100)).await;
+            self.ioctl_set_u32(110, 0, 1).await; // SET_GMODE = auto
+            self.ioctl_set_u32(142, 0, 0).await; // SET_BAND = any
 
-        self.ioctl_set_u32(110, 0, 1).await; // SET_GMODE = auto
-        self.ioctl_set_u32(142, 0, 0).await; // SET_BAND = any
+            Timer::after(Duration::from_millis(100)).await;
 
-        Timer::after(Duration::from_millis(100)).await;
+            self.state_ch.set_ethernet_address(mac_addr);
+        }
 
-        self.state_ch.set_ethernet_address(mac_addr);
+        if bluetooth_enabled {
+            // TODO: call runner.init_bluetooth somehow and pass it bluetooth_firmware?
```

Comment:
I assume this is here because it has to be done after `load_clm`? 

If there will more bluetooth commands in the future (like turning it off, changing some settings) that are not HCI, you could use a similar construct as the Ioctl module. 

This is basicalily how the interaction works. You allocate some space for the response, send an event to the runner. And then wait for the runner to call your waker once the response to the operation is ready.